### ISSUE TITLE
bug/MDX file is not recognized as markdown

### ISSUE THREAD

**Issue by MKhalusova** [Unstructured-IO/unstructured-python-client#127]
**Describe the bug**
When passing a *.mdx file to the API with unstructured client, it is not recognized as supported. 
It works, however, with open source library, both with `partition()` function and `partition_md()`

Error:

```
ERROR:unstructured-client:Failed to partition the document.
API error occurred: Status 400
{"detail":"File type None is not supported."}
```

**To Reproduce**

```
from unstructured_client.models import shared
from unstructured_client.models.errors import SDKError
from unstructured_client import UnstructuredClient

client = UnstructuredClient(
    api_key_auth=os.getenv("UNSTRUCTURED_API_KEY"),
    server_url=os.getenv("UNSTRUCTURED_URL"),
)
filename = "/content/access.mdx"

with open(filename, "rb") as f:
    files=shared.Files(
        content=f.read(),
        file_name=filename,
    )

req = shared.PartitionParameters(
    files=files
)

try:
    resp = client.general.partition(req)
    md_elements = resp.elements
except SDKError as e:
    print(e)
    md_elements = []
```

**Expected behavior**
The API should recognize a *.mdx file as markdown and process accordingly.

**Comment by awalker4** [comment]
The API just got a filetype handling fix and I verified that .mdx is working now! I'll keep this open until the new logic is deployed to Serverless.

**Comment by awalker4** [comment]
Fixed in the 1.0.42 api release!
